proto/batch_plan.proto

```diff
@@ -58,12 +58,18 @@ message ScanRange {
 }
 
 message SourceNode {
+  enum SourceType {
+    UNSPECIFIED = 0;
+    SCAN = 1;
+    COUNT_STAR = 2;
+  }
   uint32 source_id = 1;
   repeated plan_common.ColumnCatalog columns = 2;
   map<string, string> with_properties = 3;
   repeated bytes split = 4;
   catalog.StreamSourceInfo info = 5;
   map<string, secret.SecretRef> secret_refs = 6;
+  SourceType source_type = 7;
```

Comment:
I think this new field is too special to be a part of the `SourceNode` proto because it only meaningful for iceberg. Considering we will try to support predicate pushdown for iceberg scan, I think we can separate iceberg scan from source node just like file scan. https://github.com/risingwavelabs/risingwave/blob/2e1d910bc18b5a9351d961de92cb8b832b52ddad/src/frontend/src/optimizer/plan_node/batch_file_scan.rs#L80